### PR TITLE
DOC: Update scipy_proc.json with 2022 conference dates and names

### PR DESCRIPTION
**Pull request by mepa**
This PR updates the `dev` branch with 2022 conference dates and names, first before creation of the 2022 branch.
- Issue: https://github.com/scipy-conference/scipy_proceedings/issues/671

### CODE REVIEW

**Comment by scoobies** [comment]
Hi!, I'm @scoobies, a friendly bot.

Type ```@scoobies help``` to discover how I can help you.

File: scipy_proc.json [hunk 1]
```diff
@@ -195,11 +191,7 @@
       "members": [
         {
           "org": "HEB",
-          "name": "Paul Anze"
-        },
-        {
-          "org": "Oregon State University",
-          "name": "Kyle Neimeyer"
+          "name": "Paul Anzel"
```
Comment (deniederhut) [inline scipy_proc.json:194]: I think Paul works at Codecov now. But oof at me spelling his name wrong in 2019 😩

File: scipy_proc.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "proceedings": {
-    "doi": "",
+    "doi": "10.25080/majora-1b6fd038-02b",
```
Comment (deniederhut) [inline scipy_proc.json]: This gets populated by the build process (it changes every year) so it's fine to leave blank

File: scipy_proc.json (hunk 1 above)
Comment (mepa) [reply scipy_proc.json:194]: Thanks! Updated Paul's org.

File: scipy_proc.json (hunk 2 above)
Comment (mepa) [reply scipy_proc.json]: Removed

**Review by deniederhut — APPROVED** [review]
Thanks Meghann!

**Pull request merged**